### PR TITLE
Enabling SMTP feature to send emails via ActionMailer

### PR DESCRIPTION
**Pull request by smarlaku820**
- Invoking the `UserMailer` mailer via the Application Controller
- So, far the SMTP email templates have been added around reviews
- SMTP notifications will be triggered to the `project_admins` CCing the `user` requesting a review.
- `user` will be notified if the review request is `approved` or `revoked`
- If the Admins request for more changes, the `user` will also be notified of the same.

Attaching the sample screenshots

**Welcome Email**

welcome_user_smtp

**Request Review**

request_review_smtp

**Request Additional Changes**

request_review_changes_smtp

**Approve Request**

approve_review_smtp

**Revoke Request**

revoke_review_smtp

### CODE REVIEW

**Review by vanessuniq — CHANGES_REQUESTED** [review]

File: app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    @current_user, @component_id, @comment, @rule = args
+    @stig_id = "#{Component.find(@component_id).prefix}-#{@rule.rule_id}"
+    @project_id = Component.find(@component_id).project.id
+    @project_admins = get_project_admins(@project_id)
+  end
+
+  def parse_mailer_welcome_user_args(*args)
+    @current_user, @membership = args
+    membership_id = @membership.membership_id
+    @project_id = membership_id
```
[the remark this answers is not in the document]
Comment (smarlaku820) [reply app/mailers/user_mailer.rb]: Hi @vanessuniq 
created an instance variable `@project_name` for reuse in the template.

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  end
+
+  def parse_mailer_review_args(*args)
+    @current_user, @component_id, @comment, @rule = args
+    @stig_id = "#{Component.find(@component_id).prefix}-#{@rule.rule_id}"
+    @project_id = Component.find(@component_id).project.id
+    @project_admins = get_project_admins(@project_id)
+  end
+
+  def parse_mailer_welcome_user_args(*args)
```
[the remark this answers is not in the document]
Comment (smarlaku820) [reply app/mailers/user_mailer.rb:105]: Yep, now based on the `membership_type` we trigger two different SMTP notifications.

**Vulcan Project Access SMTP**
vulcan_project_access

**Vulcan Component Access SMTP**
vulcan_component_access

**Comment by aaronlippold** [comment]
https://github.com/excid3/noticed?? useful?

File: app/mailers/user_mailer.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (smarlaku820) [reply app/mailers/user_mailer.rb]: hey @vanessuniq 
reduced the round trips now. pushed fresh changes.

**Comment by smarlaku820** [comment]
@aaronlippold , 

Thank you for suggesting the library. Although I haven't reviewed it yet, I'm open to considering it since it appears to offer extensive support for various delivery methods, which could be beneficial for our project

**Comment by vanessuniq** [comment]
> https://github.com/excid3/noticed?? useful?

This could solve several delivery methods in one implementation. @smarlaku820 what do you think? I'll try to play with it.

**Review by vanessuniq — APPROVED** [review]

**Comment by vanessuniq** [comment]
> @aaronlippold ,
> 
> Thank you for suggesting the library. Although I haven't reviewed it yet, I'm open to considering it since it appears to offer extensive support for various delivery methods, which could be beneficial for our project

@smarlaku820  Should we wait on this PR, try to use the gem Aaron suggested to see how it works out??

**Comment by aaronlippold** [comment]
I always like it when others work through the hard stuff and push it into a lib for me :)

**Comment by smarlaku820** [comment]
@vanessuniq 

I haven't got the bandwidth for the next couple of weeks. Maybe, I will take a look at the lib after that.

**Pull request merged**